dcos_launch/platforms/arm.py

```diff
@@ -223,28 +227,41 @@ def wait_for_deployment(self, timeout=60 * 60):
         """
         log.info('Waiting for deployment to finish')
 
+        def azure_failure_report():
+            deploy_ops = self.azure_wrapper.rmc.deployment_operations.list(
+                    self.group_name, DEPLOYMENT_NAME.format(self.group_name))
+            failures = [(op.properties.status_code, op.properties.status_message) for op
+                        in deploy_ops if op.properties.provisioning_state == 'Failed']
+            for failure in failures:
+                log.error('Deployment operation failed! {}: {}'.format(*failure))
+
         @retrying.retry(
             wait_fixed=60 * 1000, stop_max_delay=timeout * 1000,
             retry_on_result=lambda res: res is False,
             retry_on_exception=lambda ex: isinstance(ex, CloudError))
         def check_deployment_operations():
             deploy_state = self.get_deployment_state()
+
             if deploy_state == 'Succeeded':
                 return True
+
             elif deploy_state == 'Failed':
-                log.info('Deployment failed. Checking deployment operations...')
-                deploy_ops = self.azure_wrapper.rmc.deployment_operations.list(
-                    self.group_name, DEPLOYMENT_NAME.format(self.group_name))
-                failures = [(op.properties.status_code, op.properties.status_message) for op
-                            in deploy_ops if op.properties.provisioning_state == 'Failed']
-                for failure in failures:
-                    log.error('Deployment operation failed! {}: {}'.format(*failure))
-                raise Exception('Deployment Failed!')
+                log.info('Deployment failed. Checking deployment operations.')
+                azure_failure_report()
+                raise DeploymentError('Azure Deployment Failed!')
```

Comment:
For consistency with the rest of dcos-launch I would use this error format:
https://github.com/dcos/dcos-launch/blob/master/dcos_launch/arm.py#L55

Reply:
Hey @cprovencher  and @margaret  - Using `LauncherError` does not seem right to me.
It is not a **LauncherError* (like it missed ssh keys or a field was invalid). I have moved `DeploymentError` to a `util.py`.

Also, on the format of current DeploymentError. The existing format is new to me.

* It is not calling Super()

The right way to do Custom Exception is simply inherit from Exception and if you are doing anything futher, call super() for the result of the flow.

More information here 

* https://stackoverflow.com/a/1319675
* https://www.programiz.com/python-programming/user-defined-exception
* https://julien.danjou.info/python-exceptions-guide/